### DISCUSSION TITLE
Unable to transfer playback to raspotify

### DISCUSSION THREAD

**Discussion by abrecht** [dtcooper/raspotify#526] [Q&A] [answered]
Hi,
I'm trying to control raspotify via the spotify web api. (btw. is there any way to control playback directly?)
That works fine when the same spotify user was already playing back on the pi. But after playback was paused for a while, playback will fail (404, device not found). Listing the currently active devices will not return the raspotify device id.
Only after I connect to raspotify with my phone for a second, the transfer request will succeed again.
What to do?
Basically, I'm failing to build a standalone player on my raspberry pi. Any idea how this could be achieved?
Thank you
Manuel

**Comment by JasonLG1979** [comment, marked as answer]
1st of all Raspotify is just a packaged version of librespot. Technical questions are probably best answered there.  

> (btw. is there any way to control playback directly?)

No. The intention of librespot is to be a very simple Spotify Connect client. It is designed to be controlled externally. librespot-java has a control API though.

> That works fine when the same spotify user was already playing back on the pi. But after playback was paused for a while, playback will fail (404, device not found). Listing the currently active devices will not return the raspotify device id.
Only after I connect to raspotify with my phone for a second, the transfer request will succeed again.
What to do?

Not much you can do. It's probably a limitation of the current implementation. Librespot is the product of reverse engineering some things work better than others as a result. The latest API version is in the process of being deciphered in the `new-api` librespot branch after it is merged into the `dev` branch and proves to be functional Raspotify will follow. Until then you can build your own binary and see if that solves your problem.

**Reply by abrecht** [reply]
Thank you for the quick response!
